src/player.cpp

```diff
@@ -3394,7 +3394,8 @@ void Player::doAttacking(uint32_t)
 		if (!classicSpeed) {
 			setNextActionTask(task, false);
 		} else {
-			g_scheduler.addEvent(task);
+			g_scheduler.stopEvent(classicAttackEvent);
```

Comment:
Don't you have to check if the `classicAttackEvent` is even valid before trying to stop it?

Reply:
> Don't you have to check if the `classicAttackEvent` is even valid before trying to stop it?

No. If it's not set yet by script, it's set to `0` in `player.h` and `stopEvent` code checks, if it's not 0:
```cpp
void Scheduler::stopEvent(uint32_t eventId)
{
	if (eventId == 0) {
		return;
	}
...
```